### PR TITLE
CrowdStrike Falcon Streaming v2 - remove stream refresh functionality and use session client timeout

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] In Progress

## Description
- removed stream refresh functionality
- added client session timeout to be a random time of 25 minutes +/- 5 minutes
- increased stream discovery timeout to 30 seconds

### CODE REVIEW

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 1]
```diff
@@ -149,69 +143,78 @@ async def fetch_event(
...
+            except TimeoutError as e:
+                demisto.debug(f'Failed discovering stream: {e} - '
+                              f'Going to sleep for 30 seconds and then retry - {traceback.format_exc()}')
+                await sleep(30)
+            else:
+                demisto.debug('Done waiting for stream discovery')
+                events_fetched = 0
+                new_lines_fetched = 0
+                last_fetch_stats_print = datetime.utcnow()
+                client_timeout = MINUTES_25 + uniform(-5 * 60, 5 * 60)
```
Comment (anara123) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:160]: I would say lets random between 15-25 minutes to be on the safe side, and not 20-30. 30 is the maximum limit.

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 2]
```diff
@@ -55,17 +56,24 @@ def set_auth_headers(self, token: str) -> None:
...
-    def refresh_stream_session(self) -> None:
+        while True:
+            try:
+                return self._http_request(
+                    method='GET',
+                    url_suffix='/sensors/entities/datafeed/v2',
+                    params={'appId': self.app_id},
+                )
+            except DemistoException as e:
+                if 'Error in API call [401]' in str(e):
```
Comment (anara123) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:69]: lets add some sleep here. 0-10 secs

Comment (Itay4) [reply Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:70]: @anara123 done

File: Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py [hunk 3]
```diff
@@ -55,17 +56,27 @@ def set_auth_headers(self, token: str) -> None:
...
+                    sleep_time = uniform(1, 10)
+                    demisto.debug(f'Got status code 401 on stream discovery, going to sleep for {sleep_time} - {str(e)}')
```
Comment (anara123) [inline Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:71]: is there any chance we are writing to logs `demisto.debug` in parallel?

Comment (Itay4) [reply Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py:71]: didn't encounter such issue in that integration in any env before

**Review by anara123 — APPROVED** [review]

**Pull request closed without merging**